code/implementation/docker/machine/src/main/java/io/cattle/platform/docker/machine/launch/MachineDriverLoader.java

```diff
@@ -0,0 +1,130 @@
+package io.cattle.platform.docker.machine.launch;
+
+import static io.cattle.platform.core.model.tables.MachineDriverTable.*;
+import io.cattle.platform.archaius.util.ArchaiusUtil;
+import io.cattle.platform.core.model.MachineDriver;
+import io.cattle.platform.core.model.tables.records.MachineDriverRecord;
+import io.cattle.platform.json.JsonMapper;
+import io.cattle.platform.lock.LockCallbackNoReturn;
+import io.cattle.platform.lock.LockManager;
+import io.cattle.platform.object.ObjectManager;
+import io.cattle.platform.object.process.ObjectProcessManager;
+import io.cattle.platform.object.process.StandardProcess;
+import io.cattle.platform.util.type.InitializationTask;
+
+import java.io.InputStream;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.netflix.config.DynamicStringProperty;
+
+public class MachineDriverLoader implements InitializationTask {
+
+    public static final DynamicStringProperty CONFIG_FILE = ArchaiusUtil.getString("machine.driver.config");
+
+    private static final Logger log = LoggerFactory.getLogger(MachineDriverLoader.class);
+
+    @Inject
+    LockManager lockManager;
+    @Inject
+    ObjectManager objectManager;
+    @Inject
+    ObjectProcessManager processManager;
+    @Inject
+    JsonMapper jsonMapper;
+
+    @Override
+    public void start() {
+        // do logic once
+        CONFIG_FILE.addCallback(new Runnable() {
+            @Override
+            public void run() {
+                loadDrivers();
+            }
+        });
+        loadDrivers();
```

Comment:
`loadDrivers` should really throw an exception if something fails.  On first start this will fail the start of rancher server.  But on reload (`addCallback`) the exception should be caught and ignored.